eks/mng/scale/scale.go

```
@@ -0,0 +1,215 @@
+		zap.Duration("initial-wait", update.InitialWait),
+	)
+	select {
+	case <-time.After(update.InitialWait):
+		ts.cfg.Logger.Info("waited, starting MNG scaler",
+			zap.String("cluster-name", ts.cfg.EKSConfig.Name),
+			zap.String("mng-name", mngName),
+			zap.Int("asg-min-size", cur.ASGMinSize),
+			zap.Int("asg-max-size", cur.ASGMaxSize),
+			zap.Int("asg-desired-capacity", cur.ASGDesiredCapacity),
+			zap.Int64("target-min-size", update.MinSize),
+			zap.Int64("target-max-size", update.MaxSize),
+			zap.Int64("target-desired-size", update.DesiredSize),
+		)
+	case <-ts.cfg.Stopc:
+		ts.cfg.Logger.Warn("MNG scaler wait aborted; exiting", zap.String("mng-name", mngName))
+		return errors.New("MNG scaler wait aborted")
+	}
+
+	// ref. https://docs.aws.amazon.com/cli/latest/reference/eks/update-nodegroup-config.html
+	if err = ts.scaleMNG(mngName, update); err != nil {
+		return err
+	}
+
+	ts.cfg.Logger.Info("completed MNG scaler",
+		zap.Int("from", cur.ASGDesiredCapacity),
+		zap.Int64("to", update.DesiredSize),
+	)
+	return nil
+}
+
+// Scales MNG to the next scaling configuration in the array.
+func (ts *tester) scaleMNG(mngName string, update *eksconfig.MNGScaleUpdate) (err error) {
+	cur := ts.cfg.EKSConfig.AddOnManagedNodeGroups.MNGs[mngName] 
+	nodegroupConfigInput := &aws_eks.UpdateNodegroupConfigInput{
+		ClusterName:   aws.String(ts.cfg.EKSConfig.Name),
+		NodegroupName: aws.String(mngName),
+		ScalingConfig: &aws_eks.NodegroupScalingConfig{DesiredSize: aws.Int64(update.DesiredSize),
+			MaxSize: aws.Int64(update.MaxSize),
+			MinSize: aws.Int64(update.MinSize)}}
+	var updateOut *eks.UpdateNodegroupConfigOutput
+	updateOut, err = ts.cfg.EKSAPI.UpdateNodegroupConfig(nodegroupConfigInput)
+
+	var reqID string
+	if err != nil {
+		ts.cfg.Logger.Warn("MNG scaler request failed", zap.String("mng-name", mngName), zap.Error(err))
+		return err
+	}
+	if updateOut.Update != nil {
+		reqID = aws.StringValue(updateOut.Update.Id)
+	}
```

Can we return an error if request ID is `""`?